File: ch13_3.m
clear all; close all; clc
% hw 6,  CH13 -3 data 
ABO = [7 5 3 13; 27 32 8 18; 55 50 7 24];
MN = [21 6 1 ; 54 27 5; 74 51 11];

%% conduct tests
% compute chisq stats for ABO type
df.ABO = prod(size(ABO) - 1);
Eij.ABO = (sum(ABO,2) * sum(ABO,1))/sum(ABO(:));
testStats.ABO = sum(sum(((ABO - Eij.ABO).^2) ./Eij.ABO));
pval.ABO = 1 - chi2cdf(testStats.ABO,df.ABO);

% compute chisq stats for MN type 
df.MN = prod(size(MN) - 1);
Eij.MN = (sum(MN,2) * sum(MN,1))/sum(MN(:));
testStats.MN = sum(sum(((MN - Eij.MN).^2) ./Eij.MN));
pval.MN = 1 - chi2cdf(testStats.MN,df.MN);

%% display test stats and crit 
testStats
pval
chi2inv(0.95,df.ABO)
chi2inv(0.95,df.MN)

File: ch13_3.R
rm(list=ls(all=TRUE))
setwd('/Users/Qihong/Dropbox/github/STAT310_MathematicalStatistics')
# load data


# a) plot diff against control 
diff = data$X.test. - data$X.control.
plot(diff~data$X.control.)
# => if the control rate is high, then the difference become more negative, 
# the magnitude also increases

# b) mean diff, sd, CI
Xbar = mean(data$X.test.)
Ybar = mean(data$X.control.)
varX = sum((data$X.test. - Xbar)^2)/(n-1)
varY = sum((data$X.control. - Ybar)^2)/(n-1)
SeX = sqrt(varX/n) 
SeY = sqrt(varY/n)

meanDiff = mean(data$X.test.) - mean(data$X.control.)
varDiff = (varX + varY - cor(data$X.test.,data$X.control.) * sqrt(varX*varY))/n
sdDiff = sqrt(varDiff)

meanDiff
sdDiff

# CI = meanDiff +- sdDiff * T
CI_low = meanDiff - sdDiff * qt(0.95,n-1)
CI_high = meanDiff + sdDiff * qt(0.95,n-1)


#c) conduct a t test, since the variance is unknown
# under null: 
testStats = meanDiff / (sdDiff / sqrt(n))
qt(0.999,n-1)

# test statistics > crit
# conclusion: reject 